platform/core/src/services/HangingProtocolService/lib/validator.js

```diff
@@ -1,69 +1,231 @@
+ * @example
+ * value = ['abc', 'def', 'GHI']
+ * testValue = 'abc' (Fail)
+ *           = ['abc'] (Fail)
+ *           = ['abc', 'def', 'GHI'] (Valid)
+ *           = ['abc', 'GHI', 'def'] (Fail)
+ *           = ['abc', 'def'] (Fail)
+ *
+ * value = 'Attenuation Corrected'
+ * testValue = 'Attenuation Corrected' (Valid)
+ *           = 'Attenuation' (Fail)
+ *
+ * */
+validate.validators.equals = function(value, options, key) {
+  const testValue = getTestValue(options);
+
+  // If options is an array, then we need to validate each element in the array
+  if (Array.isArray(testValue)) {
+    // If the array has only one element, then we need to compare the value to that element
+    if (testValue.length === 1) {
+      if (testValue[0] !== value) {
+        return `${key} must equal ${testValue[0]}`;
+      }
+    } else if (testValue.length !== value.length) {
+      return `${key} must be an array of length ${testValue.length}`;
+    }
+    // We need to compare each element in the array
+    else {
+      for (let i = 0; i < testValue.length; i++) {
+        if (testValue[i] !== value[i]) {
+          return `${key} ${testValue[i]} must equal ${value[i]}`;
+        }
+      }
+    }
   }
-};
 
+  // If options is not an array, we can compare the value directly
+  else if (value !== testValue) {
+    return `${key} must equal ${testValue}`;
+  }
+};
+/**
+ * check if the value is not equal to options
+ *
+ * @example
+ * value = ['abc', 'def', 'GHI']
+ * testValue = 'abc' (Valid)
+ *           = ['abc'] (Valid)
+ *           = ['abc', 'def', 'GHI'] (Fail)
+ *           = ['abc', 'GHI', 'def'] (Valid)
+ *           = ['abc', 'def'] (Valid)
+ *
+ * value = 'Attenuation Corrected'
+ * testValue = 'Attenuation Corrected' (Fail)Valid
+ *           = 'Attenuation' (Valid)
+ *
+ * */
 validate.validators.doesNotEqual = function(value, options, key) {
-  const testValue = options?.value ?? options;
-  if (value === testValue) {
-    return key + 'cannot equal ' + testValue;
+  const testValue = getTestValue(options);
+
+  if (Array.isArray(testValue)) {
+    if (testValue.length === 1) {
+      if (testValue[0] === value) {
+        return `${key} must not be ${testValue}`;
+      }
+    } else {
+      if (testValue.length === value.length) {
+        let score = 0;
+        testValue.forEach((x, i) => {
+          if (x === value[i]) {
+            score++;
+          }
+        });
+        if (score === testValue.length) {
+          return `${key} must not equal to ${testValue}`;
+        }
+      }
+    }
+  } else if (value === testValue) {
+    return `${key} must not equal to ${testValue}`;
   }
 };
 
+/**
+ * Check if a value includes one or more specified options.
+ *
+ * @example
+ * value = ['abc', 'def', 'GHI']
+ * testValue = ‘abc’ (Valid)
+ *           = ‘dog’ (Fail)
+ *           = [‘att’, ‘abc’] (Valid)
+ *           = ['abc', 'def', 'dog'] (Valid)
+ *           = ['cat', 'dog'] (Fail)
+ *
+ * value = 'Attenuation Corrected' (Fail)
```

Comment:
This case should be equivalent to ['Attenuation Corrected'], which should pass if the testValue contains the same identical string.